### ISSUE TITLE
Crash on initialization on samsung device

### ISSUE THREAD

**Issue by kevincmojo** [AppsFlyerSDK/appsflyer-android-sdk#44]
We're seeing a crash related to appsflyer on a certain samsung device. Our setup is pretty simple:

```
        val appsFlyer = AppsFlyerLib.getInstance()
        appsFlyer.setDebugLog(true)
        appsFlyer.init("<dev-key">, null, application.applicationContext)
        appsFlyer.setAppId("<app-id>")
```

It seems to work on most devices but we've noticed a crash that we are unable to prevent because of the nature of how it happens. We are not seeing a stack trace in the logs but instead it looks like a content provider the sdk is connecting to (powered by samsung's digitial turbine ignite) is forcing the OS to kill our app.

```
01-06 16:43:13.502  1329  1396 I ActivityManager: Start proc 11195:com.dti.samsung/u0a243 for content provider {com.dti.samsung/com.digitalturbine.igniteui.provider.AppsFlyerContentProvider}

01-06 16:43:14.582 11195 11357 D ConnectivityManager: StackLog: [android.net.ConnectivityManager.unregisterNetworkCallback(ConnectivityManager.java:4835)] [com.digitalturbine.ignite.jobs.reachability.ConnectivityCallback.disable(SourceFile:3)] [com.digitalturbine.ignite.jobs.reachability.ReachabilityManager.stop(SourceFile:7)] [com.digitalturbine.ignite.controller.IgniteController.scheduleNextAndSuicide(SourceFile:1)] [com.digitalturbine.igniteui.igniteconnector.IgniteConnectorImpl$b.actionsForEvent(SourceFile:100)] [com.digitalturbine.ignite.event.EventActionManager.onEvent(SourceFile:3)] [com.digitalturbine.ignite.event.EventManager.c(SourceFile:5)] [com.digitalturbine.ignite.event.EventManager.b(SourceFile:1)] [z6.run(Unknown Source:4)]
01-06 16:43:14.593 11195 11357 I com.dti.samsung: System.exit called, status: 0
01-06 16:43:14.593 11195 11357 I AndroidRuntime: VM exiting with result code 0, cleanup skipped.
01-06 16:43:14.612 11110 11156 D BF/Bugfender-SDK: There's a new Bugfender SDK version. Please check bugfender.com.
01-06 16:43:14.616  1329  1787 I StatusBarManagerService: binder died for pkg=com.dti.samsung
01-06 16:43:14.617  1329  2299 I ActivityManager: Process com.dti.samsung (pid 11195) has died: fg  BTOP(94,1531)
01-06 16:43:14.618  1329  1397 I libprocessgroup: Successfully killed process cgroup uid 10243 pid 11195 in 1ms
01-06 16:43:14.619   938   938 I Zygote  : Process 11195 exited cleanly (0)

01-06 16:43:14.619  1329  2299 I ActivityManager: Killing 11110:our.package.name/u0a348 (adj 0): depends on provider com.dti.samsung/com.digitalturbine.igniteui.provider.AppsFlyerContentProvider in dying proc com.dti.samsung (adj -10000)
```

Looks like the Appsflyer sdk connects to AppsFlyerContentProvider which is bound to `com.dti.samsung`. At some point either the content provider or the app `com.dti.samsung` is killed and because the appsflyer sdk holds a reference to the content provider client, it kills our app as well. It seems that instead the sdk should be referencing the content provider using an unstable client to prevent the app from dying if the content provider is killed. https://stackoverflow.com/questions/33423825/how-and-when-to-use-a-contentproviderclient-obtained-using-acquireunstableconten/33562256#33562256.

**Comment by JaerenMoneyBox** [comment]
Hey - we are having the same issue as the above since upgrading from `6.1.0` to `6.9.4`, specifically on SM-A536B devices.

Is there any update on the above?

**Comment by epool** [comment]
We are also facing this same issue.

Sorry for tagging you guys @sokoloff06 @af-fess @pazlavi but the repo issues seem kinda orphan.

Do you have a clue/workaround/fix for this? Thank you in advance! 🙏🏼

**Comment by pazlavi** [comment]
Hey @epool, 
We are aware of previous occurrences of this issue.
In order to determine if this is the same case on your end, we recommend submitting a ticket to our support team at support@appsflyer.com.
Our support team will be able to guide you with the required steps based on your case.

**Comment by pazlavi** [comment]
In January 2023 we started receiving reports regarding system crashes in several Samsung device models that were referring to the AppsFlyer SDK as the source of the crash. The issue is occurring in Samsung [AER devices](https://www.android.com/enterprise/recommended/) (Android Enterprise Recommended, usually used by organizations managing employee devices).
 
We conducted a thorough investigation and discovered that the crash was the result of a 3rd party service that is preinstalled on the device: This service was terminating prematurely, causing the AppsFlyer SDK to crash and the app to stop working.
 
Today, we released a new SDK version - `6.10.1` that solves the crash that happens by 3rd party service’s behavior.
